contrib/vcloud/util.py

```diff
@@ -0,0 +1,90 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import collections
+import os
+import sys
+
+import benchexec.util
+
+sys.dont_write_bytecode = True  # prevent creation of .pyc files
+
+
+def parse_vcloud_run_result(values):
+    result_values = collections.OrderedDict()
+
+    def parse_time_value(s):
+        if s[-1] != "s":
+            raise ValueError('Cannot parse "{0}" as a time value.'.format(s))
+        return float(s[:-1])
+
+    def set_exitcode(new):
+        if "exitcode" in result_values:
+            old = result_values["exitcode"]
+            assert (
+                old == new
+            ), "Inconsistent exit codes {} and {} from VerifierCloud".format(old, new)
+        else:
+            result_values["exitcode"] = new
+
+    for key, value in values:
+        value = value.strip()
+        if key in ["cputime", "walltime"]:
+            result_values[key] = parse_time_value(value)
+        elif key == "memory":
+            result_values["memory"] = int(value.strip("B"))
+        elif key == "exitcode":
+            set_exitcode(benchexec.util.ProcessExitCode.from_raw(int(value)))
+        elif key == "returnvalue":
+            set_exitcode(benchexec.util.ProcessExitCode.create(value=int(value)))
+        elif key == "exitsignal":
+            set_exitcode(benchexec.util.ProcessExitCode.create(signal=int(value)))
+        elif (
+            key in ["host", "terminationreason", "cpuCores", "memoryNodes", "starttime"]
+            or key.startswith("blkio-")
+            or key.startswith("cpuenergy")
+            or key.startswith("energy-")
+            or key.startswith("cputime-cpu")
+        ):
+            result_values[key] = value
+        elif key not in ["command", "timeLimit", "coreLimit", "memoryLimit"]:
+            result_values["vcloud-" + key] = value
+
+    return result_values
+
+
+def parse_frequency_value(s):
```

Comment:
This is duplicated from `benchexec/utils.py`. Probably it is good to check what else is duplicated in the `util` module.

Reply:
This does not raises an exception. I wonder if that was the reason to maintain two copies.

Reply:
Hm, both methods contain `raise ValueError`? If there are differences in behavior, we should still deduplicate the common parts.

Reply:
You have more knowledge on this than me. I don't know why was the code duplicated originally.
If you want I can remove the call and use the function from the `benchexec.util`

Reply:
I just noticed that the copy in `vcloudutil.py` uses `KHz` instead of `kHz` (which is a wrong unit), and I am not sure whether the unit handling is otherwise identical. So we would need to check which format and unit the VerifierCloud actually produces and then we can decide whether the copy in BenchExec correctly parses this.

Reply:
I am also not sure. I also think that I am not the right person to address this.

Reply:
@PhilippWendler actually you might be better positioned to answer this.
This function was introduced by you in [this commit](https://gitlab.com/sosy-lab/software/cpachecker/-/commit/508661e56ec6f385dd64a339abec229e4a9891b1)
and in benchexec utils it was introduced by @TBunk in 494e6009b006c9ccaaf49cef5c364e10692ae8c3

As I said before, I am happy to call whatever of the two is better. But I don't think that I should be making this decision.

Reply:
Well, I also have to check what the VerifierCloud actually outputs and compare that against the two implementations of this method, it is not easier for me.

But I did so know, found a bug in the method (6750c9663febde492761d0c4111dfefb3bc0c503), and found out why the standard BenchExec method for this cannot be used (f71a4ca03b8c7c63816bab57a71f9d28a7139e2d).